### PR TITLE
image-partition: Modify property 'fsuuid' to allow setting UUID

### PR DESCRIPTION
**Pull request by sangorrin**
Currently support setting UUID for btrfs, xfs, ext2, ext3, ext4 file system

Signed-off-by: Nguyen Thi Huong <huong4.nguyenthi@toshiba.co.jp>
Signed-off-by: Daniel Sangorrin <daniel.sangorrin@toshiba.co.jp>

### CODE REVIEW

**Comment by sangorrin** [comment]
Hi,

This patch adds functionality that allows you to specify the UUID of your file system instead of getting a random one. Some use cases include reproducible builds and software updates.

Kind regards,
Daniel

File: actions/image_partition_action.go [hunk 1]
```diff
@@ -319,12 +327,22 @@ func (i ImagePartitionAction) formatPartition(p *Partition, context debos.DebosC
 		cmdline = append(cmdline, "mkfs.hfsplus", "-s", "-v", p.Name)
 		// hfsx is case-insensitive hfs+, should be treated as "normal" hfs+ from now on
 		p.FS = "hfsplus"
+	case "xfs":
```
Comment (obbardc) [inline actions/image_partition_action.go:330]: split xfs into separate commit/PR

File: actions/image_partition_action.go [hunk 2]
```diff
@@ -145,6 +149,7 @@ import (
 	"fmt"
 	"github.com/docker/go-units"
 	"github.com/go-debos/fakemachine"
+	"github.com/google/uuid"
```
Comment (obbardc) [inline actions/image_partition_action.go:152]: what's the uuid lib used for ?

File: actions/image_partition_action.go [hunk 3]
```diff
@@ -624,6 +642,16 @@ func (i *ImagePartitionAction) Verify(context *debos.DebosContext) error {
 		case "":
 			return fmt.Errorf("Partition %s missing fs type", p.Name)
 		}
+		if len(p.FSUUID) > 0 {
```
Comment (obbardc) [inline actions/image_partition_action.go:645]: this should be checked earlier ?

**Review by obbardc — CHANGES_REQUESTED** [review]
thanks! just some small changes required :-)

File: actions/image_partition_action.go [hunk 4]
```diff
@@ -88,6 +89,9 @@ for partition.
 - fsck -- if set to `false` -- then set fs_passno (man fstab) to 0 meaning no filesystem
 checks in boot time. By default is set to `true` allowing checks on boot.
 
+- fsuuid -- file system UUID string. This option is only supported for btrfs,
+ext2, ext3, ext4, and xfs.
```
Comment (obbardc) [inline actions/image_partition_action.go:93]: nit: remove extra comma before and

File: actions/image_partition_action.go (hunk 2 above)
Comment (obbardc) [reply actions/image_partition_action.go:152]: right, to check if the uuid is correct. Can't seem to resolve these comments for some reason :(

**Review by obbardc — CHANGES_REQUESTED** [review]

**Comment by sangorrin** [comment]
Sorry for the delay

I was preparing a repo for testing and better commits (including answers to your questions that I guessed should be in the commits as well).

I also added xfsprogs to the Dockerfile.

You can find this repository for testing the fsuuid if you want:
- https://github.com/sangorrin/debos-tests-fsuuid

I will reply to your comments now!!

Thanks,
Daniel

File: actions/image_partition_action.go (hunk 1 above)
Comment (sangorrin) [reply actions/image_partition_action.go:330]: OK, done. Sorry about that.

File: actions/image_partition_action.go (hunk 2 above)
Comment (sangorrin) [reply actions/image_partition_action.go:152]: Sorry for the delay.

The uuid library is used for verifying the input value of the FSUUID property (specified by the user on the image-partition action). The verification is based on RFC 4122 and DCE 1.1: Authentication and Security Services. Check
https://github.com/google/uuid for more details.

I have added this information to the commit as well, just in case someone wants to know while looking at the commit log.

File: actions/image_partition_action.go (hunk 3 above)
Comment (sangorrin) [reply actions/image_partition_action.go]: We have moved the check a bit earlier but still within the verify function. Is that OK for you or did you mean something else?

File: actions/image_partition_action.go (hunk 4 above)
Comment (sangorrin) [reply actions/image_partition_action.go:93]: OK, just a second

Comment (sangorrin) [reply actions/image_partition_action.go]: Done!

**Comment by sangorrin** [comment]
OK, I have now replied to all your comments. Sorry for the delay.
